### DISCUSSION TITLE
Folytatás

### DISCUSSION THREAD

**Discussion by marcelltoth** [marcelltoth/NeptunLight#9] [General]
@batizisanya @Plank99 @daem-on @66Gramms @SgtVoro illetve bárki aki szeretne résztvenni a projekt jövőjében

## 👋 Sziasztok

Először is elnézést kérek, hogy nem reagáltam a bejövő PR-ekre, pedig nagyon hálás vagyok értük, főleg, hogy a projekt Github repoját nem reklámoztam sehol, még egy README-je sincs, ti pedig így is megtaláltátok valahogy, és be akartok segíteni. Köszönöm.

Ennek legfőbb oka az, hogy én már jó pár éve nem vagyok egyetemista, illetve Xamarinnal egyébként sem foglalkoztam soha hobbiszinten túl, és hát a munka meg egyebek mellett nem fér bele nekem ennek az appnak a fejlesztgetése. Play-ből is unlisteltem, Nektek akiknek már megvolt nem fogja letörölni, de újak már nem látják.

Az aláíró private key szerencsére megvan még, de nem csináltam release-t évek óta, szerintem fel sincs telepítve amire szükség van (ha még emlékszem egyáltalán).

## ⚠️ Ismert problémák

A fentieken túl van  az appban 1-2 ismert probléma is, tudom, amin mindenképpen kéne dolgozni. Ezek között a könnyebbek az URL változások amikre Ti pl a PR-eket küldtétek, de hallottam, hogy egyes egyetemeken pl. CAPTCHA-s login van, na azt megcsinálni kemény feladat lenne. Illetve persze az egyész ugye egy nem publikus APIt használ (konkrétan scrapel), így az is nyilván folyamatosan változik. Egy szó, mint száz, van feladat bőven, és mindig is lesz.

## ❓Hogyan tovább?

Mint mondtam, nekem sajnos nem lesz időm ezzel foglalkozni, de ha Nektek van, annak nagyon örülök, és természetesen megpróbálok segíteni. Kérdés az, hogy mit gondoltok jónak, hogy legyen tovább? 

Az én belátásom szerint itt a következő lehetőségek vannak, a legdrasztikusabbtól a legenyhébbig:
  - Valaki leforkolja a repot, kiad egy új appot. Itt igazából csak egy LICENCE-t kéne hozzáadjak aztán kész.
  - Valaki leforkolja a repot, de én átadom a signing key-t és felveszem Play-en, így ugyanarra a névre tud új verziókat publikálni.
  - Megpróbálok csinálni valami CD pipelinet ami `master`-re merge után automatikusan publishol, a kód marad ebben a repoban, de felveszek 1-2 embert collaboratornak, hogy lehessen review-zni és mergelni nélkülem is. 

Az utóbbi talán a legkényelmesebb mindenkinek gondolom, bár azzal van plusz munkám az elején, ha gondoljátok azt bevállalom. De ha valamelyik másik verzió tetszik, vagy egy negyedik ötlet amire nem gondoltam, mondjátok, én nyitott vagyok mindenre.

**Mit gondoltok**, mi lenne a jó megoldás? Illetve ki vállalná, hogy core maintainer lesz, annak minden felelősségével?

**Comment by 66Gramms** [comment]
A CI-s megoldással nekem az a baj, hogy nem biztos, hogy minden master merge után kéne publisholni, illetve szerintem neked sem lenne jó ha extra munkát kéne csinálnod.

Nekem a második megoldás tűnik talán a legjobbnak

**Comment by marcelltoth** [comment]
Lehet amúgy kombinálni is a dolgot: a CI/CD csak draft verziót tölt fel, és valaki(k)nek lesz joga publikálni kézzel. (Én másik saját appnál ezt csináltam, úgy is, hogy egyedül voltam, kényelmes.) 

Nem vagyok én ellenére annak, hogy valakire rábízni teljesen, csak kicsit félek, hogy ugyanolyan vége lesz mint nekem. Az ember egyszer elvégzi az egyetemet, vagy csak úgy lesz egyéb dolga és aztán kutatni kell utána. Ha meg van automatika akkor sokkal könnyebb maintainereket váltani. 

De persze azt is lehet, hogy ha találunk valakit akinek sok ideje van (és szintén ért hozzá), hogy ő leforkolja, aztán majd csinál CD-t magának a forkjában.

**Reply by 66Gramms** [reply]
Ez a kombinált megoldás nekem inkább hangzik jónak :)

**Comment by daem-on** [comment]
Örülök, hogy továbbra is foglalkozol ezzel pedig gondolom sok dolgod van. Én ahogy a PR-emben említettem nem értek (és nem nagyon szeretnék érteni) a Xamarinhoz, így az első két lehetőséghez nem sok hozzászólásom van. A harmadikat azért vetettem fel, mert elég önző módon ezzel nekem nem lenne semmi dolgom, kisebb változtatásokat, mint az URL-t bárki egyszerűen átírhatja és nem kell az egész környezetet felállítaniuk hozzá. De tudom hogy ebben sok munka van, nem biztos hogy a legjobb megoldás.

Azt tudom mondani, ha lenne egy CI/CD felállítva én szívesen tesztelném Androidon publikálás előtt.

**Comment by marcelltoth** [comment]
Megpróbálkozom feléleszteni a dev környezetemet és alkotni valami CI/CD-t. Nem ígérek semmit előre, de próbálkozom. 😄 🤞

**Comment by palmarci** [comment]
Sziasztok! A pull requesteket kijavítottam a kódban, és készítettem egy APK-t. A Visual Studio szépen feltelepített mindent amit magának kellett, majd gyakorlatilag hiba nélkül lefordult. Sikerült belépnem a BME-s fiókomba, viszont az üzenetek és a kurzusok nem mennek.
Ha valakit érdekel:
[neptunlight_2023_09_13.zip](https://github.com/marcelltoth/NeptunLight/files/12601649/neptunlight_2023_09_13.zip)
